_posts/2017-07-14-tutorial1-ritotjs-basic-component.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+title:  "Riot Tutorial 1 - Componente Básico"
+date:   2017-07-14 10:00:00 -0300
+tags: tutorial
+categories: en
+comments: true
+---
+
+En este tutorial vamos a crear un componente o etiqueta personalizada
+utilizando [riot.js](http://riotjs.com/){:target="\_blank"} el cuál mostrará un
+mensaje de saludo con un nombre previamente establecido, o con la palabra
+"mundo" sino se suministra ningún nombre.
+
+## Componente
+Los componentes **pueden incluír** tres secciones:
+1. **CSS** (estilos del componente).
+2. **HTML**.
+3. **Javascript** (comportamiento deseado).
+
+
+```html
+<!-- hello-world.tag -->
+<hello-world>
+  <style>
+    h1 { color: red; }
+  </style>
+
+  <h1>hola {name}!</h1>
+
+  <script>
+    name = opts.name
```

Comment:
Bueno acá hay dos cosas para aclarar:
1. El "this" lo estuvimos discutiendo con @olvap si iba o no. Yo lo borré porque en la guía de riot decía:

> Smart ES6 like method syntax is supported: methodName() { } becomes this.methodName = function() {}.bind(this) where this always points to the current tag instance.

Pero puede que lo haya entendido mal y se refería a cuando estás **usando** del tag en el HTML, no en el archivo .tag. Bueno, si está mal me avisan y lo vuelvo a poner.

2.  Es cierto @eloyesp. Lo hicimos así porque @olvap me enseñó que se pueden pasar atributos al componente en el momento de montarlo. La idea sería mostrar ese concepto.